src/hotspot/share/gc/shared/collectedHeap.hpp

```diff
@@ -513,6 +517,11 @@ class CollectedHeap : public CHeapObj<mtInternal> {
 
   virtual bool is_oop(oop object) const;
 
+  // Historic gc information
+  size_t get_heap_free_at_last_gc()    { return _heap_capacity_at_last_gc - _heap_used_at_last_gc; }
+  size_t get_heap_used_at_last_gc()    { return _heap_used_at_last_gc; }
+  void update_heap_info_at_gc();
+
```

Comment:
Maybe move these to after unused(). Alternatively, move it near soft_ref_policy(), since this is currently only used by soft ref handling.